napari/layers/_shapes_layer/model.py

```diff
@@ -444,22 +499,30 @@ def block_update_properties(self):
     def _get_shape(self):
         """Determines the shape of the vertex data.
         """
-        if len(self.data._vertices) == 0:
-            return [1, 1]
+        if len(self._data_view._vertices) == 0:
+            slice_shape = [1, 1]
+        else:
+            slice_shape = list(np.max(self._data_view._vertices, axis=0) + 1)
+
+        if len(self.data.keys()) == 1:
```

Comment:
I suggest adding some comments here for what it means for `len(self.data.keys()) == 1`. It's not immediately obvious what this translates to in terms of the shape of the shapes layer.

Reply:
added